flink-runtime/src/main/java/org/apache/flink/streaming/api/windowing/assigners/SlidingEventTimeWindows.java

```diff
@@ -71,7 +71,8 @@ public Collection<TimeWindow> assignWindows(
         if (timestamp > Long.MIN_VALUE) {
             List<TimeWindow> windows = new ArrayList<>((int) (size / slide));
             long lastStart = TimeWindow.getWindowStartWithOffset(timestamp, offset, slide);
-            for (long start = lastStart; start > timestamp - size; start -= slide) {
+            long lower = timestamp - size;
```

Comment:
nit: can we add final to this and `long lastStart`

Reply:
Done!